cluster/gce/templates/create-salt-files.sh

```diff
@@ -33,21 +35,19 @@ enable_cluster_dns: '$(echo "$ENABLE_CLUSTER_DNS" | sed -e "s/'/''/g")'
 dns_replicas: '$(echo "$DNS_REPLICAS" | sed -e "s/'/''/g")'
 dns_server: '$(echo "$DNS_SERVER_IP" | sed -e "s/'/''/g")'
 dns_domain: '$(echo "$DNS_DOMAIN" | sed -e "s/'/''/g")'
+gce_node_names: '$(echo "$KUBERNETES_NODE_NAMES" | sed -e "s/'/''/g")'
 EOF
 
-mkdir -p /srv/salt-overlay/salt/nginx
-echo $MASTER_HTPASSWD > /srv/salt-overlay/salt/nginx/htpasswd
+# This should only happen on cluster initialization
+if [[ ! -z "${KUBELET_TOKEN:-}" ]]; then
+  mkdir -p /srv/salt-overlay/salt/nginx
+  echo $MASTER_HTPASSWD > /srv/salt-overlay/salt/nginx/htpasswd
 
-# Generate and distribute a shared secret (bearer token) to
-# apiserver and kubelet so that kubelet can authenticate to
-# apiserver to send events.
-# This works on CoreOS, so it should work on a lot of distros.
-kubelet_token=$(cat /dev/urandom | base64 | tr -d "=+/" | dd bs=32 count=1 2> /dev/null)
+  mkdir -p /srv/salt-overlay/salt/kube-apiserver
+  known_tokens_file="/srv/salt-overlay/salt/kube-apiserver/known_tokens.csv"
+  (umask u=rw,go= ; echo "${KUBELET_TOKEN},kubelet,kubelet" > $known_tokens_file)
```

Comment:
I think `umask 077` is more clear in its intent. In ~15 years with Linux this is the first time I see this kind of usage...

Also, double quotes around the `${known_tokens_file}` variable.